app.py

```diff
@@ -224,23 +250,51 @@ def skill():
     current data and then return the JSONified object 
     '''
     if request.method == 'GET':
-        index = request.args.get("index")
-        if index is not None:
-            return get_skill_by_index(data, index)
-        return jsonify(data["skill"])
+        return handle_get_skill()
 
     if request.method == 'POST':
+        return handle_post_skill()
+
+    if request.method == 'PUT':
         req = request.get_json()
 
-        required_fields = {"name":"string", "proficiency":"string", "logo":"string"}
+        choice = request.args.get("corrections")
+        if choice == 'yes':
+            corrections = spell_check(req)
+            if corrections:
+                return jsonify(corrections)
 
-        code, err_message = validate_request(req, required_fields)
+    return jsonify({"Server Error": "Couldn't process method"})
 
-        if code != 0:
-            return jsonify({"error": err_message}), code
+def handle_get_skill():
+    '''
+    Handles skill get request
+    '''
+    index = request.args.get("index")
```

Comment:
I love that you modularized the code, well done!

There is something I would like to suggest, pass the params directly to the functions `handle_get_skill(request.args.get('index'))` to avoid conflicts

Reply:
I wanted to maintain the consistency of other fellows' code. There's also a check `if index is not None`. The alternative will be to write something like `if (request.args.get('index')) is not None`. I think the first style is more readable and also maintains the code consistency, while the second style is more ambiguous. Please, let me know what you think about this